src/debugging/DockerServerReadyAction.ts

```diff
@@ -289,15 +289,17 @@ class MultiOutputDockerServerReadyManager extends vscode.Disposable implements D
 }
 
 class DockerDebugAdapterTrackerFactory implements vscode.DebugAdapterTrackerFactory {
-    private static trackers: Map<vscode.DebugSession, MultiOutputDockerServerReadyManager> = new Map<vscode.DebugSession, MultiOutputDockerServerReadyManager>();
+    private static trackers: Map<string, MultiOutputDockerServerReadyManager> = new Map<string, MultiOutputDockerServerReadyManager>();
 
     public static start(session: vscode.DebugSession): DockerDebugAdapterTracker | undefined {
         const configuration = <ResolvedDebugConfiguration>session.configuration;
         if (configuration?.dockerOptions?.dockerServerReadyAction) {
-            let tracker = DockerDebugAdapterTrackerFactory.trackers.get(session);
+            const realSessionId = configuration?.__sessionId as string || session.id;
```

Comment:
Consider using a utility function to calculate `realSessionId` (the same expression shows up twice in the code)